test/test_basic.py

```diff
@@ -213,9 +213,10 @@ async def test_correct_passphrase(self, key_dir: pathlib.Path, runtime_dir: path
         assert 'keys/id_rsa.enc' in prompt
 
     @pytest.mark.asyncio
-    @pytest.mark.xfail
     async def test_known_host_good(self, key_dir: pathlib.Path, runtime_dir: pathlib.Path) -> None:
-        await self.run_test(key_dir, runtime_dir, ZeroDivisionError(), 'passphrase',
+        # this calls do_hostkey() for the already known key, just in case it wants to supply additional keys
+        # don't do this and don't accept any, just rely on the existing one
+        await self.run_test(key_dir, runtime_dir, False, 'passphrase',
                             handle_host_key=True, known_host_key='scan')
```

Comment:
For actually fixing cockpit, I have to get rid of setting `{Global,User}KnownHostsFile=none` with handle_host_keys=True, as they ruin the show. Dropping them does not break any test.

However, *this* test here really ought to fail right now, as without any existing database there is no way that the SSH client can change an existing host key. I think this somehow interferes with setting another known_hosts file in the test, as *that's* the one that has the bad key. This does not model reality well, so let's fix this. Back to draft.

Reply:
Added a separate commit for that.